nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/functions/ToBytes.java

```
@@ -0,0 +1,85 @@
+ * limitations under the License.
+ */
+package org.apache.nifi.record.path.functions;
+
+import org.apache.nifi.record.path.FieldValue;
+import org.apache.nifi.record.path.RecordPathEvaluationContext;
+import org.apache.nifi.record.path.StandardFieldValue;
+import org.apache.nifi.record.path.paths.RecordPathSegment;
+import org.apache.nifi.record.path.util.RecordPathUtils;
+import org.apache.nifi.serialization.record.RecordFieldType;
+import org.apache.nifi.serialization.record.util.DataTypeUtils;
+
+import java.nio.charset.Charset;
+import java.util.stream.Stream;
+
+public class ToBytes extends RecordPathSegment {
+
+    private final RecordPathSegment recordPath;
+    private final RecordPathSegment charsetSegment;
+
+    public ToBytes(final RecordPathSegment recordPath, final RecordPathSegment charsetSegment, final boolean absolute) {
+        super("toBytes", null, absolute);
+        this.recordPath = recordPath;
+        this.charsetSegment = charsetSegment;
+    }
+
+    @Override
+    public Stream<FieldValue> evaluate(RecordPathEvaluationContext context) {
+        final Stream<FieldValue> fieldValues = recordPath.evaluate(context);
+        return fieldValues.filter(fv -> fv.getValue() != null)
+                .map(fv -> {
+
+                    if (!(fv.getValue() instanceof String)) {
+                        return fv;
+                    }
+
+                    final Charset charset = getCharset(this.charsetSegment, context);
+
+                    final byte[] bytesValue;
+                    try {
+                        Byte[] src = (Byte[]) DataTypeUtils.toArray(fv.getValue(), fv.getField().getFieldName(), RecordFieldType.BYTE.getDataType(), charset);
+                        bytesValue = new byte[src.length];
+                        for(int i=0;i<src.length;i++) {
+                            bytesValue[i] = src[i];
+                        }
+                    } catch (final Exception e) {
+                        return fv;
```

Should probably let the Exception fly - as-is, it is just silently swallowing the Exception and then returning an object of the wrong type.